src/transformers/utils/hub.py

```diff
@@ -1123,6 +1124,31 @@ def move_cache(cache_dir=None, new_cache_dir=None, token=None):
         )
 
 
+class PushInProgress:
+    """
+    Internal class to keep track of a push in progress (which might contain multiple `Future` jobs).
+    """
+
+    def __init__(self, jobs=None):
```

Comment:
```suggestion
    def __init__(self, jobs: Optional[Future] = None) -> None:
```

Even if not putting types everywhere, IMO having it here makes it explicit to the developer of which Future we are talking about. Must be imported with `from concurrent.futures import Future`, in opposition to `asyncio.Future` that also exists.